### PR TITLE
Refactored Main Parts of Code

### PR DESCRIPTION
**Pull request by emmasax1**
Description and Impact
----------------------
No impact for production. Simply refactored code in order to build up our Code Climate rating, reduce complexity in our methods, reduce duplicated code, fix little bugs.

Deploy Plan
-----------
No migrations present.

Rollback Plan
-------------
* To roll back this change, revert the merge with: `git revert -m 1 MERGE_SHA` and perform another deploy.

URLs
----
> Links to bug tickets or user stories.

QA Plan
-------
- [ ] Play around on staging to make sure everything still works
- [ ] Make sure tests pass
- [ ] Check code style and quality

### CODE REVIEW

File: app/assets/javascripts/hubstats/bootstrap.js [hunk 1]
```diff
@@ -4,7 +4,7 @@
  * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
  */
```
Comment (emmasax1) [inline app/assets/javascripts/hubstats/bootstrap.js]: I added all of the semi colons because in Javascript, although it technically doesn't need semicolons everywhere, it can be helpful so that the compiler knows for sure where the developer wants to end its statements.... and Code Climate hated all of the missing semicolons. I went through line by line with Code Climate and added semicolons where they pointed out one was missing.

File: app/controllers/hubstats/application_controller.rb [hunk 2]
```diff
@@ -9,5 +9,12 @@ def set_time
       @timespan = TIMESPAN_ARRAY[cookies[:hubstats_index].to_i][:date].ago.to_date
     end
 
+    def grouping (group_request, group)
```
Comment (bbergstrom) [inline app/controllers/hubstats/application_controller.rb]: Delete this duplicate.

File: app/controllers/hubstats/deploys_controller.rb [hunk 3]
```diff
@@ -75,5 +58,32 @@ def create
         end
       end
     end
+
+    def valid_repo(repo)
+      if !repo.empty?
+        @deploy.repo_id = repo.first.id.to_i
```
Comment (bbergstrom) [inline app/controllers/hubstats/deploys_controller.rb]: Having hte assignment in here is a bit misleading.  I like the validity function, but consider moving the assignment back under the if statement.

File: app/controllers/hubstats/deploys_controller.rb [hunk 4]
```diff
@@ -75,5 +58,32 @@ def create
...
+    def valid_pr_ids
+      if !@pull_request_id_array.empty? && @pull_request_id_array != [0]
+        @deploy.pull_requests = Hubstats::PullRequest.where(repo_id: @deploy.repo_id).where(number: @pull_request_id_array)
+        return true
+      end
+      return false
+    end
+
+    def valid_pulls
+      if !@deploy.pull_requests.first.nil?
```
Comment (bbergstrom) [inline app/controllers/hubstats/deploys_controller.rb]: Consider drying this up a bit, something like below should work.
```
pull = @deploy.pull_requests.first
return false if pull.nil? || pull.merged_by.nil?
@deploy.user_id = pull.merged_by
return true
```

File: app/controllers/hubstats/pull_requests_controller.rb [hunk 5]
```diff
@@ -9,8 +9,6 @@ def index
       pull_ids = Hubstats::PullRequest
         .belonging_to_users(params[:users])
         .belonging_to_repos(params[:repos])
-        .state_based_order(@timespan,params[:state],"ASC")
-        .map(&:id)
```
Comment (bbergstrom) [inline app/controllers/hubstats/pull_requests_controller.rb]: Why was this removed?

Comment (emmasax1) [inline app/controllers/hubstats/pull_requests_controller.rb]: The ordering isn't necessary because the labels call below orders what's we get from the pull_ids. The map line was a mistake. I'll fix it.

**Comment by emmasax1** [comment]
@bbergstrom Any last comments on this PR?

**Comment by bbergstrom** [comment]
:+1:
